client/app/helper/controller-helper.service.js

```diff
@@ -40,6 +40,26 @@ class ControllerHelper {
             window.open(`data:text/plain;${charSet},${dataString}`);
         }
     }
+
+    /**
+     * copies given message to clipboard
+     * @param {string} messageToCopy, message to copy to clipboard
+     * @return {any} error if copy failed, empty string otherwise
+     */
+    copyToClipboard (messageToCopy) {
+        try {
+            const dummy = document.createElement("input");
+            document.body.appendChild(dummy);
+            dummy.setAttribute("id", "dummy_id");
+            dummy.setAttribute("value", messageToCopy);
+            dummy.select();
+            document.execCommand("copy");
+            document.body.removeChild(dummy);
```

Comment:
only in firefox and chrome :)

Reply:
Ok could you leave a note to test it on IE / Edge / Safari when you send your JIRA to QA please?

Reply:
Also as I stated before, please apply visibility = "hidden" to your dummy :).  I think we can't see it because of angular's digest cycle.. however I'd had visibility = "hidden" anyway, just to be safe.

Reply:
sure I will do that .. its good to be double safe

Reply:
i cannot set visibility to hidden. If I set that I will not be able to focus element to copy text.
Below code does not work if element is not in focus.
 dummy.select();
 document.execCommand("copy");